src/twitter/Timeline.js

```diff
@@ -240,50 +282,59 @@ class Timeline {
       this._newtags.delete(req.user_id);
     }
 
-    this.twitter.storage.get_since("@"+tline.twitter_id).then((since) => {
-      log.silly("Polling %s, since value: %s", "@"+tline.twitter_id, since);
-      if (since) {
-        req.since_id = since;
-      }
-      return this.twitter.client_factory.get_client();
-    }).then((client)=>{
-      return client.getAsync('statuses/user_timeline', req).catch((error) =>{
-        log.error("Timeline", "_process_timeline: GET /statuses/user_timeline returned: %s", error.code);
-      });
-    }).then((feed) => {
-      if (feed.length === 0) {
-        return;
-      }
-      else if(feed.length === TIMELINE_TWEET_FETCH_COUNT) {
-        log.info("Timeline poll request hit count limit. Request likely incomplete.");
+    const since = yield this.twitter.storage.get_since("@"+tline.twitter_id);
+    if (since) {
+      req.since_id = since;
+    }
+    let feed;
+    try {
+      feed = yield client.getAsync('statuses/user_timeline', req);
+    }
+    catch (error) {
+      log.error("Timeline", "_process_timeline: GET /statuses/user_timeline returned: %s", error.code);
+      return;
+    }
+    if (feed.length === 0) {
+      return;
+    }
+    else if(feed.length === TIMELINE_TWEET_FETCH_COUNT) {
+      log.info("Timeline", "Poll request hit count limit. Request likely incomplete.");
+    }
+    const s = feed[0].id_str;
+    this.twitter.storage.set_since("@"+tline.twitter_id, s);
+    // If req.count = 1, the resp will be the initial tweet used to get initial "since"
+    if (req.count !== 1) {
+      try {
+        yield this.twitter.processor.process_tweets(tline.room, feed, TWEET_REPLY_MAX_DEPTH);
       }
-      const s = feed[0].id_str;
-
-      log.silly("Timeline", "Storing since: %s", s);
-      this.twitter.storage.set_since("@"+tline.twitter_id, s);
-
-      // If req.count = 1, the resp will be the initial tweet used to get initial "since"
-      if (req.count !== 1) {
-        this.twitter.processor.process_tweets(tline.room, feed, TWEET_REPLY_MAX_DEPTH);
+      catch(err) {
+        log.error("Timeline", "Error whilst processing timeline %s: %s", tline.twitter_id, err);
       }
-
-      tline.hasProcessedTweets = true;
-    }).catch((err) => {
-      log.error("Timeline", "Error whilst processing timeline %s: %s", tline.twitter_id, err);
-    });
-
-    this._t++;
-    if(this._t >= this._timelines.length) {
-      this._t = 0;
     }
   }
 
-  _process_hashtags () {
-    if (this._hashtags.length < 1) {
+  * _process_hashtags () {
```

Comment:
Maintaining near-identical code-paths for hashtags/timelines is going to get really annoying.